### PR TITLE
[concurrency] Implement swift_task_runOnMainActor/hop to main actor thunk

### PR DESCRIPTION
**Pull request by gottesmm**
[concurrency] Implement swift_task_runOnMainActor.

This routine takes a synchronous non-throwing main actor isolated closure
without a result. If we are dynamically on the main actor, we just run the
closure synchronously. Otherwise, we run a new task on the main actor and call
the closure on that.

This builds on top of the previous commit by using
swift_task_isCurrentExecutor_v2 in the implementation of this function.

To backwards deploy this function on Darwin, I used some tricks from libdispatch
to validate that we are on the main queue.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
I ripped out the backwards compatibility stuff. There is more to it and we may need some more compiler functionality or a compatibility library.

**Comment by gottesmm** [comment]
but this is just for a prototype, so it makes sense to just get it done.

File: stdlib/public/Concurrency/Task.swift [hunk 1]
```diff
@@ -1259,3 +1259,53 @@ internal func _runTaskForBridgedAsyncMethod(@_inheritActorContext _ body: __owne
...
 
 #endif
+
+#if !SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY && !SWIFT_CONCURRENCY_EMBEDDED
+
+@available(SwiftStdlib 9999, *)
+@_silgen_name("swift_task_isCurrentExecutor_v2")
+@usableFromInline
+internal func _taskIsCurrentExecutor(
+  executor: Builtin.Executor, flags: UInt64) -> Bool
```
Comment (ktoso) [inline stdlib/public/Concurrency/Task.swift:1269]: Can we put this one immediately next to the existing overload please?

**Review by ktoso — APPROVED** [review]
Yeah this is nicer now, by reusing the existing infra -- now it'd be great if we can get an SPI to make this checking more reliable -- Thanks for reusing the existing infra 👍

**Review by ktoso — COMMENTED** [review]
I would prefer we rename the `swift_task_isCurrentExecutor_v2` -> `swift_task_isCurrentExecutorWithFlags` though to follow existing conventions, if this were to be merged.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci toolchain

**Comment by gottesmm** [comment]
Just going to do hop to main actor thunk in this as well.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@swift-ci toolchain

**Comment by gottesmm** [comment]
@swift-ci build toolchain

**Pull request merged**